.travis.yml

```diff
@@ -16,5 +16,6 @@ jobs:
     - stage: Test
       script:
         - mvn test -Pcoverage
-        - 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then sonar-scanner; fi'
+        - 'if [ "$TRAVIS_PULL_REQUEST" = "false" AND "$SONAR_TOKEN" != ""]; then sonar-scanner; fi'
         - bash <(curl -s https://codecov.io/bash)
+        - curl -s -X POST -H "Content-Type:application/json" -H "Accept:application/json" -H "Travis-API-Version:3" -H "Authorization:token $TRAVIS_API_TOKEN" -d '{"request":{"branch":"master"}}' https://api.travis-ci.org/repo/project-xavier%2Fxavier-integration/requests
```

Comment:
@jonathanvila I suggest moving the location of the Trigger command to `after_success` as you can see in this PR https://github.com/jonathanvila/xavier-analytics/pull/5. `after_success` will be executed at the end of all Jobs, this way we don't mix test commands and the Trigger command. YOu can see the result of this change here https://travis-ci.org/carlosthe19916/xavier-analytics/builds/636404098

I also think we can add `if [ "$TRAVIS_PULL_REQUEST" = "false"` to the trigger command because `xavier-integration` will always use `xavier-analytics` master branch and it is not useful to trigger a new build from a PR (in xavier-analytics) if this PR is not going to be used by `xavier-integration`. What do you think?

Reply:
I agree with both suggestions.

Reply:
> I tried to check the build triggered in `xavier-integration` project for the build of your last commit in this PR but i didn't find it.
> In Travis CI, there's an error re the credentials https://travis-ci.org/project-xavier/xavier-analytics/builds/636306556#L5785

@mrizzi I think the error is because the build is coming from a Pull Request, and PRs don't have access to Environment Variables because of security restrictions. It is the same problem why we decided to omit Sonar analysis in PRs.

We should add a condition to execute the Trigger command, something like `if [ "$TRAVIS_PULL_REQUEST" = "false"`

Reply:
@carlosthe19916 thanks for the clarification.
It would have been great to be able to test a PR in `xavier-analytics` using the e2e test in `xavier-integration` to be sure that the changes in the PR itself were working fine with the current code.
I understand that PRs' builds has this security restriction so it seems there's sadly nothing to do.